Updated docs for ROCm 7.x.x (#239)
ROCm-Developer-Tools/ROC_SHMEM: README.md

implementation. This forwarding of requests is transparent to the
programmer, who only sees the GPU-side interface.

> The GPU Direct Async (GDA) backend allows for rocSHMEM to construct NIC operations directly

Reviewer: ```suggestion
The GPU Direct Async (GDA) backend allows for rocSHMEM to issue communication operations to the NIC directly from the device-side code, without involving a CPU proxy.
```

within the GPU.

> During initialization we prepair network resources for each NIC vendor using their

Reviewer: ```suggestion
During initialization we prepare network resources for each NIC vendor using the vendor-appropriate
```

Direct Verbs APIs.

> At runtime, we construct the relevent Work Queue Entries (WQEs) and post them our send queues.

Reviewer: ```suggestion
When calling the device-side rocSHMEM API, the device threads are used to construct Work Queue Entries (WQEs) and post the communication to the send queues of the NIC directly.
```

> We also allow for device-side polling of our Compleation Queues (CQs).

Reviewer: ```suggestion
Completion Queues (CQs) are polled from the device-side code as well.
```

Other MPI implementations, such as MPICH,
_should_ be compatible with rocSHMEM but it has not been thoroughly tested.

## Building the Dependencies using our Helper Script

> We have script to install dependencies.

Reviewer: ```suggestion
We have a script to install dependencies.
```

However, it is not guaranteed to work and perform optimally on all platforms.
Configuration options are platform dependent.

```
BUILD_DIR=/path/to/not_rocshmem_src_or_build/dependencies /path/to/rocshmem_src/sripts/install_dependencies.sh
```

After compiling and installing UCX and Open MPI, please update your `PATH` and `LD_LIBRARY_PATH`
to point to the installation locations.

> The exact locations will be printed if the script is run successfully.

Reviewer: ```suggestion
The exact locations will be printed if the script ran successfully.
```
